yw_website/yw_website/apps/website/serializers.py

```diff
@@ -157,25 +282,186 @@ def __create_tags(self, w, validated_data):
                     tw = TagWorkflow(tag=t, workflow=w)
                     tw.save()
 
-    def __create_scripts(self, v, validated_data): 
-        scripts = ScriptSerializer(validated_data.get('scripts'), many=True)
+    def _create_scripts(self, v, validated_data):
+        scripts = ScriptSerializer(validated_data.get("scripts"), many=True)
         for script in scripts.data:
-            s = Script(name=script.get('name'), version=v, checksum=script.get('checksum'), content=script.get('content'))
+            s = Script(
+                name=script.get("name"),
+                version=v,
+                checksum=script.get("checksum"),
+                content=script.get("content"),
+            )
             s.save()
-    
-    def __create_files(self, r, validated_data):
-        files = FileSerializer(validated_data.get('files'), many=True)
+
+    def _create_files(self, r, validated_data):
+        files = FileSerializer(validated_data.get("files"), many=True)
         for file in files.data:
             f, _ = File.objects.update_or_create(
-                checksum=file.get('checksum'),
-                defaults={                
-                    'size':file.get('size'),
-                    'name':file.get('name'),
-                    'uri':file.get('uri'),
-                    'last_modified':file.get('lastModified')
-                }
+                checksum=file.get("checksum"),
+                defaults={
+                    "size": file.get("size"),
+                    "name": file.get("name"),
+                    "uri": file.get("uri"),
+                    "last_modified": file.get("lastModified"),
+                },
             )
             f.save()
             rf = RunFile(run=r, file=f)
             rf.save()
 
+    def _create_program_blocks(self, r, validated_data):
+        program_blocks = ProgramBlockSerializer(
+            validated_data.get("programBlock"), many=True, context={"run": r}
+        )
+        for program_block in program_blocks.data:
+            if program_block.get('inProgramBlock') != None:
+                try:
+                    in_block = ProgramBlock.objects.get(
+                        programblock_id=program_block.get("inProgramBlock"), run=r
+                    )
+                except ObjectDoesNotExist:
+                    raise serializers.ValidationError("Parent programblock does not exist when trying to create programBlock")
```

Comment:
nice extra check

Reply:
There's two things being checked, if their is no inProgramBlock that is valid, but if the inProgramBlock does not exist already that is an error

Reply:
i know, i was saying this was a good extra check. see `_create_data` comments

Reply:
Applied to Data too now